File: in_Python/1380 Lucky Numbers in a Matrix.py
class Solution(object):
    def luckyNumbers (self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: List[int]
        """
        rowmin = []
        colmax = []
        res = []
        for r in range(len(matrix)):
            cur_row_min = matrix[r][0]
            for c in range(len(matrix[0])):
                if matrix[r][c] < cur_row_min:
                    cur_row_min = matrix[r][c]
            rowmin.append(cur_row_min)

        for c in range(len(matrix[0])):
            cur_col_max = matrix[0][c]
            for r in range(len(matrix)):
                if matrix[r][c] > cur_col_max:
                    cur_col_max = matrix[r][c]
            colmax.append(cur_col_max)
        for i in rowmin:
            if i in colmax:
                res.append(i)
                break
        return res
                
        

File: in_Java/1380 Lucky Numbers in a Matrix.java
class Solution {
    public List<Integer> luckyNumbers (int[][] matrix) {
        int row = matrix.length;
        int col = matrix[0].length;
        int[] rowmin = new int[row];
        int[] colmax = new int[col];
        int curmin;
        int curmax;
        for(int r = 0; r < row; r++){
            curmin = Integer.MAX_VALUE;
            for(int c = 0; c < col; c++){
                if(curmin > matrix[r][c]){
                    curmin = matrix[r][c];
                }
            }
            rowmin[r] = curmin;
        }
        for(int c = 0; c < col; c++){
            curmax = Integer.MIN_VALUE;
            for(int r = 0; r < row; r++){
                if(curmax < matrix[r][c]){
                    curmax = matrix[r][c];
                }
            }
            colmax[c] = curmax;
        }
        List<Integer> res = new ArrayList<>();
        List<Integer> t = new ArrayList<>();
        for(int i : colmax) t.add(i);
        for(int i : rowmin){
            if(t.contains(i)){
                res.add(i);
            }
        }
        return res;
    }
}